### ISSUE TITLE
Performance not fast enough for fast typing. Letters are recognized but it takes a second

### ISSUE THREAD

**Issue by GenosseFlosse** [florisboard/florisboard#689]
#### Short description
Fast typing leads to the input being displayed with a significant time difference. In the following screenrecording the printing of letters came in sort of "waves" instead of instantly after my tap on the letter. This certainly is not fast enough for production use on my device.
https://user-images.githubusercontent.com/59205524/114883366-26540e00-9e05-11eb-8705-a6e31265c743.mp4
Additionally, the preference app was very slow in responding to tab switching.

#### Steps to reproduce
1. For me it was just literally using the keyboard, was solved by reboot. 
Not reproducible after reboot. (Glad I captured the video before rebooting). I will keep an eye on that. Is there any way to manually trigger produce a debug log?

#### Environment information
- FlorisBoard Version: 0.3.10
- Install Source: GPlay
- Device: Pocophone F1
- Android version, ROM: Android 10, MIUI

**Comment by patrickgold** [comment]
Thanks for your bug report! The waves is explained by how the suggestions work: The shorter a current word is, the longer it takes to generate suggestions for it. This you see these waves, as your typed words are relatively long. (Btw the suggestions get reworked soon, so this should definitely be improved in the future).

> Solved after device reboot.

I think this is because after the device reboot FlorisBoard had more resources available because other apps did all their clean-up as well. This is definitely not a solution though, FlorisBoard must work without a device reboot too.

**Comment by GenosseFlosse** [comment]
Thanks for your optimism :D great fan of this keyboard and I will keep you updated in a week from now if I have had these issues again. If this behaviour occurs again I will first check the memory state of my phone...

**Comment by mortezae** [comment]
Yes it got too slow in latest version, I guess this occured after upgrading to new layout system. I don't use word suggestion feature at all but it's still slow in responding and sometimes completely freezing.

**Comment by another-sapiens** [comment]
When this happens, I temporarily fix it by force-stopping FlorisBoard. It takes a minute, but it's faster than a reboot.

**Comment by GenosseFlosse** [comment]
Ok I encounter the same issue again. 
Florisboard uses almost 500MByte of RAM. That's huge... How can I help you debug this?

![IMG_20210419_143232.jpg](https://user-images.githubusercontent.com/59205524/115236813-1e091500-a11c-11eb-8236-c468d36f402c.jpg)

**Comment by GenosseFlosse** [comment]
> Ok I encounter the same issue again.
> Florisboard uses almost 500MByte of RAM. That's huge... How can I help you debug this?
> 
> ![IMG_20210419_143232.jpg](https://user-images.githubusercontent.com/59205524/115236813-1e091500-a11c-11eb-8236-c468d36f402c.jpg)

Doesn't this link to #677?

**Comment by Glitchy-Tozier** [comment]
Is this still an issue? (=Can you still reproduce this?)
Big performance-improvements were made during the last months.

**Comment by patrickgold** [comment]
In general it shouldn't, but suggestions are currently not implemented, which were always a performance eater. So I guess we have to see how well the native implementation works to see if performance is decent enough so FlorisBoard can be used on all devices (I still suspect that older devices have some performance issues)

**Comment by pepmts** [comment]
I have the same behaviour of keys shown a little time after typing.
I have checked that the ram usage is not too high (55MB), I have rebooted, without success.
I think I have this problem since I moved from 0.3.14-beta5 or 0.3.14-beta7 (I don't remember I'm sorry...)

I'm on Samsung A5 2016, currently testing [v0.3.14-rc04](https://github.com/florisboard/florisboard/releases/tag/v0.3.14-rc04)

**Comment by pouek** [comment]
Same here but only on etherpad-lite (online collaborative text editor) sites. So I guess it's not coming from the keyboard.

**Comment by lm41** [comment]
I cannot reproduce this on my devices with the release builds. Debug builds are known to lag a bit.
Since this issue was opened with v0.3.10 and the latest version is `v0.6.0-alpha01` I wanted to revisit this issue and check if this problem still persists.
